paddle/fluid/framework/data_set.cc

```diff
@@ -116,6 +116,11 @@ void DatasetImpl<T>::SetMergeByInsId(
   keep_unmerged_ins_ = keep_unmerged_ins;
 }
 
+template <typename T>
+void DatasetImpl<T>::SetBoxPSFlag() {
+  boxps_flag_ = true;
```

Comment:
not a good idea to expose special PS knowledge to a common toolkit, i prefer async_close_channel_flag